### PR TITLE
nightly_version_assignment reference error when nightlypg param is missin in pkgvars

### PR DESCRIPTION
**Pull request by gurkanindibay**
pg_auto_failover packaging pkgvars does not include nightlypg. When not exists, releasepg is used as nightlypg

### CODE REVIEW

**Review by hanefi — COMMENTED** [review]
I think it makes sense for us to use dotenv to read and manipulate `pkgvars` contents. Doing the changes manually is prone to errors IMO.

https://github.com/theskumar/python-dotenv

**Comment by gurkanindibay** [comment]
> I think it makes sense for us to use dotenv to read and manipulate `pkgvars` contents. Doing the changes manually is prone to errors IMO.
> 
> https://github.com/theskumar/python-dotenv

Fixed as you suggested

File: packaging_automation/citus_package.py [hunk 1]
```diff
@@ -162,30 +163,28 @@ def sign_packages(base_output_path: str, sub_folder: str, secret_key: str, passp
...
-            if release_version_assignment is None or "=" not in release_version_assignment or len(
-                    release_version_assignment.split("=")) != 2:
-                raise ValueError(
-                    f"Release version in pkglatest is not well formatted. Expected format: releasepg=12,13 "
-                    f"Actual Format:{release_version_assignment}")
-            if nightly_version_assignment is None or "=" not in nightly_version_assignment or len(
-                    nightly_version_assignment.split("=")) != 2:
+        pkgvars_config = dotenv_values(f"{input_files_dir}/{PKGVARS_FILE}")
+        release_versions_str = pkgvars_config['releasepg']
+        if ',' not in release_versions_str:
```
Comment (hanefi) [inline packaging_automation/citus_package.py:173]: I do not like this conditional here. Does that mean we can not actually create a package for a single PG version?

We used to do that when we need to create a single local package for testing purposes for example.

I think `releasepg=12` is a well formatted line for `pkgvars` file. What do you think?

File: packaging_automation/tests/test_citus_package.py [hunk 2]
```diff
@@ -24,8 +24,8 @@
     "debian/stretch": 4,
     "debian/buster": 4,
     "ubuntu/xenial": 2,
-    "ubuntu/bionic": 2,
-    "ubuntu/focal": 2
+    "ubuntu/bionic": 4,
+    "ubuntu/focal": 4
```
Comment (hanefi) [inline packaging_automation/tests/test_citus_package.py:28]: Why did these lines change now? Did we suddenly start to create debug packages? Why is `debian/xenial` still 2?

Comment (gurkanindibay) [reply packaging_automation/tests/test_citus_package.py:28]: https://github.com/citusdata/packaging/issues/732

File: packaging_automation/citus_package.py (hunk 1 above)
Comment (gurkanindibay) [reply packaging_automation/citus_package.py]: Removed ',' validation

File: packaging_automation/citus_package.py [hunk 3]
```diff
@@ -160,32 +161,19 @@ def sign_packages(base_output_path: str, sub_folder: str, secret_key: str, passp
...
-            if nightly_version_assignment is None or "=" not in nightly_version_assignment or len(
-                    nightly_version_assignment.split("=")) != 2:
-                raise ValueError(
-                    f"Nightly version in pkglatest is not well formatted. Expected format: nightlypg=12,13 "
-                    f"Actual Format:{nightly_version_assignment}")
-            release_versions = release_version_assignment.split("=")[1].split(",")
-            nightly_versions = nightly_version_assignment.split("=")[1].split(",")
+        pkgvars_config = dotenv_values(f"{input_files_dir}/{PKGVARS_FILE}")
+        release_versions_str = pkgvars_config['releasepg']
+        if "nightlypg" in pkgvars_config:
```
Comment (hanefi) [inline packaging_automation/citus_package.py:170]: This means that we will make it optional to have a `nightlypg` key in `pkgvars`. I do not think that this was intended, but still nice to have.

File: packaging_automation/requirements.in [hunk 4]
```diff
@@ -14,4 +14,6 @@ requests
 SQLAlchemy
 urllib3
 wheel
+python-dotenv
+
```
Comment (hanefi) [inline packaging_automation/requirements.in:18]:
```suggestion
```

no need to have 2 empty lines.

File: packaging_automation/requirements.txt [hunk 5]
```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with python 3.8
+# This file is autogenerated by pip-compile
```
Comment (hanefi) [inline packaging_automation/requirements.txt:2]: I think we have different versions of `pip-compile` installed. The autogenerated comments differ on our machines. This is not an important issue I guess, as long as pip-compile does not make behaviour changes. If that happens, we may need to fix our pip-compile versions as well.

File: packaging_automation/tests/test_citus_package.py (hunk 2 above)
Comment (hanefi) [reply packaging_automation/tests/test_citus_package.py:28]: Ubuntu 16.04 EOL'd 5 years after its release: April, 2021.

We can drop packaging support, and test coverage for them, I think.

**Review by hanefi — APPROVED** [review]
Please read through my comments and create a small commit to fix formatting before merging this PR.

File: packaging_automation/tests/test_citus_package.py (hunk 2 above)
Comment (gurkanindibay) [reply packaging_automation/tests/test_citus_package.py:28]: Opened an issue for this. We have download numbers. I think first we need to check download numbers and afterwars we need to declare it to community when we will drop support https://github.com/citusdata/tools/issues/170

File: packaging_automation/requirements.in (hunk 4 above)
Comment (gurkanindibay) [reply packaging_automation/requirements.in]: Fixed

**Pull request merged**